### ISSUE TITLE
Exit codes

### ISSUE THREAD

**Issue by MacLotsen** [prantlf/nomnoml-cli#2]
When `nomnoml` gets an error it doesn't return an exit code 1.

Example source code using nomnoml:
```bash
#!/bin/bash

project_root=$(dirname ${BASH_SOURCE[0]})/..
diagram_dir=$project_root/src/diagrams/
diagram_out=$project_root/doc/res/

# GOTO root of project e.g. ~/Documents/rnd
cd $project_root

if [ ! -d $diagram_out ]; then
	mkdir -p $diagram_out
fi

gen() {
	tmp=$(basename $1)
	tmp=${tmp%.*}
	nomnoml -i $1 -o $diagram_out$tmp.png

	# Unfortunately nomnoml doesn't return a exit code on failure...
	if [ ! $? ]; then
		echo "ERROR: Failed to generate image for $tmp"
	fi
}

for diag in $diagram_dir/*.txt
do
	echo "Processing $(basename $diag)"
	gen $diag
done

echo Done
```

**Comment by prantlf** [comment]
True, it does not. When i use it, I check, if the target file exists, to detect success or failure.

I admit, that the exit code should be set as well, like command line tools usually do.

**Comment by prantlf** [comment]
Fixed by c07a2de4f54542dd8f3b4df8aa514663cc929c14 and added to [release 0.5.0](https://github.com/prantlf/nomnoml-cli/releases/tag/v0.5.0). The error message is printed on stderr in case of failure. Enjoy! :-)

**Comment by MacLotsen** [comment]
Thank you!
